base4kids2.ldif

```diff
@@ -498,6 +498,43 @@ attributeTypes: (
       DESC 'Determines whether a base4kids object is disabled but still active'
       SUP base4kidsIsBlocked )
 #
+# Name:     base4kidsTelephoneNumberBusiness
+# Syntax:   Phone number
+# Examples: +01234567890
+attributeTypes: (
+      1.3.6.1.4.1.31534.2.5.2.42
+      NAME 'base4kidsTelephoneNumberBusiness'
+      DESC 'Business telephone number'
+      EQUALITY telephoneNumberMatch
+      ORDERING telephoneNumberOrderingMatch
+      SYNTAX 1.3.6.1.4.1.1466.115.121.1.50
+      SINGLE-VALUE )
+#
+# Name:     base4kidsDateOfBirth
+# Syntax:   Generalized Time
+# Examples: 19951221000000Z
+attributeTypes: (
+      1.3.6.1.4.1.31534.2.5.2.43
+      NAME 'base4kidsDateOfBirth'
+      DESC 'Date of Birth'
+      EQUALITY generalizedTimeMatch
```

Comment:
Using the [Generalized Time](https://tools.ietf.org/html/rfc4517#section-3.3.13) syntax and the related matching and ordering rules aren't suitable for a birth date. As it will only contain the year, month and day without any time and time zone information.

I would rather suggest to store only the year, month and day according to the [ISO 8601 calendar date representation](https://en.wikipedia.org/wiki/ISO_8601#Calendar_dates).

Either use the [IA5String](https://tools.ietf.org/html/rfc4517#section-3.3.15) syntax and store the date as `YYYY-MM-DD` or use the [Numeric String](https://tools.ietf.org/html/rfc4517#section-3.3.23) syntax and store it as `YYYYMMDD` depending on your preferences.

Refer also to the following [mail thread](https://groups.google.com/forum/#!searchin/mozilla.dev.tech.ldap/$20Date$2FBirthday$20fields$20and$20LDAP$20schemas$20%7Csort:date/mozilla.dev.tech.ldap/Ast6WzQbgsQ/jD1qIS0g0BoJ).

Reply:
It's the date-time representation due to we get this value as ISO date-time from Scolaris. I wanted to keep the form up to Moddle provisioning (easier jave DT convertion + verification). Moodle gets it as specified (date only).

Reply:
@euzeoh I'm sorry but I don't understand which form (`YYYY-MM-DD` vs. `YYYYMMDD` vs. generalized time) you want to use now. Furthermore, your latest commit inherits the attribute specifications from the `base4kidsGroupType` which is unrelated to this field and might result in accidental attribute re-definition changes as you mentioned before.
The `base4kidsGroupType`  attribute has been defined as "Printable String" but you're attribute comments states "Directory String". 

I'm happy to fix an rebase everything for you, but need to know which format and syntax you would like to use - thanks.

Reply:
Hi @euzeoh

> What is the best attribute we might derrive from?

I wouldn't derive the definition of this attribute from another one as we have no date-only attribute yet and I couldn't find an existing one from a well-known schema which would make sense.


> It's been finally decided base4kidsDateOfBirth to be "YYYY-MM-DD" string

Ok, so we can't use the [Numeric String](https://tools.ietf.org/html/rfc4517#section-3.3.23) or [Integer](https://tools.ietf.org/html/rfc4517#section-3.3.16) syntax then.


> I'm not sure IA5String is the good type for this attribute due to its limitations

There will be only digits and hyphens, so you could use either [Printable String](https://tools.ietf.org/html/rfc4517#section-3.3.29) or [Directory String](https://tools.ietf.org/html/rfc4517#section-3.3.6). I would probably go with the first one as we have no need to store unicode characters.


> This is, by the way, the proper example of why we shouldn't use derrivation, as I described in another comment.

Which I agree and also didn't request it for this attribute within my first review.

Thanks